CMakeLists.txt

```diff
@@ -254,131 +272,71 @@ endif()
 #   ARMADILLO_INCLUDE_DIRS - directories necessary for Armadillo includes
 #   BOOST_ROOT - root of Boost installation
 #   BOOST_INCLUDEDIR - include directory for Boost
+#   CEREAL_INCLUDE_DIR - include directory for cereal
 #   ENSMALLEN_INCLUDE_DIR - include directory for ensmallen
 #   STB_IMAGE_INCLUDE_DIR - include directory for STB image library
 #   MATHJAX_ROOT - root of MathJax installation
-find_package(Armadillo "${ARMADILLO_VERSION}" REQUIRED)
 
+if(CMAKE_CROSSCOMPILING)
+  get_deps(https://github.com/xianyi/OpenBLAS/releases/download/v0.3.13/OpenBLAS-0.3.13.tar.gz OpenBLAS OpenBLAS-0.3.13.tar.gz)
+  if(NOT MSVC)
+    execute_process(COMMAND make TARGET=ARMV8 BINARY=64 HOSTCC=gcc CC=${CMAKE_C_COMPILER} FC=${CMAKE_FORTRAN_COMPILER} NO_SHARED=1
+                    WORKING_DIRECTORY ${CMAKE_BINARY_DIR}/deps/OpenBLAS-0.3.13)
+    file(GLOB OPENBLAS_LIBRARIES "${CMAKE_BINARY_DIR}/deps/OpenBLAS-0.3.13/libopenblas.a")
+    # set(OPENBLAS_LIBRARIES ${OPENBLAS})
+  endif()
+  find_library(GFORTRAN NAMES libgfortran.a PATHS ${CMAKE_SYSROOT}/usr/lib)
+  find_library(PTHREAD NAMES libpthread.a PATHS ${CMAKE_SYSROOT}/usr/lib)
+  set(MLPACK_LIBRARIES ${MLPACK_LIBRARIES} ${OPENBLAS_LIBRARIES} ${GFORTRAN} ${PTHREAD})
+endif()
+
+# Remove support for older 32bit system, if we need to have 32bit system
+# Since we are cross compiling, We will download it into the deps folder,
+if(DISABLE_DOWNLOADS)
+  find_package(Armadillo "${ARMADILLO_VERSION}" REQUIRED)
+elseif(NOT ARMADILLO_FOUND AND (NOT DISABLE_DOWNLOADS))
+  get_deps(http://files.mlpack.org/armadillo-10.3.0.tar.gz armadillo armadillo-10.3.0.tar.gz)
+  set(ARMADILLO_INCLUDE_DIR ${GENERIC_INCLUDE_DIR})
+  find_package(Armadillo REQUIRED)
+endif()
 # Include directories for the previous dependencies.
 set(MLPACK_INCLUDE_DIRS ${MLPACK_INCLUDE_DIRS} ${ARMADILLO_INCLUDE_DIRS})
-set(MLPACK_LIBRARIES ${MLPACK_LIBRARIES} ${ARMADILLO_LIBRARIES})
+# We do have to link with Armadillo if we are NOT cross compiling
+if(NOT CMAKE_CROSSCOMPILING)
+  set(MLPACK_LIBRARIES ${MLPACK_LIBRARIES} ${ARMADILLO_LIBRARIES})
+endif()
```

Comment:
I think that you have to add `ARMADILLO_LIBRARIES` regardless of whether or not cross-compiling is being done, since `ARMADILLO_LIBRARIES` could contain other things like HDF5, SuperLU, etc.

Reply:
Agreed but in this case, they should be header only.
I think this is the case for `HDF5`, but I am not sure for `SuperLU`